tests/patch_attribute_testslide.py

```diff
@@ -110,9 +111,17 @@ def it_fails_if_attribute_is_a_class(self):
             def with_class_attributes(context):
                 context.merge_context("patching works")
 
-        @context.xexample
+        @context.example
         def it_fails_if_new_value_is_of_incompatible_type(self):
-            pass
+            with self.assertRaises(TypeCheckError):
+                print(self.target)
+                self.patch_attribute(self.target, "typedattr", 123)
+
+        @context.example
+        def it_passes_if_new_value_is_of_incompatible_type_with_type_validation_false(
+            self,
+        ):
+            self.patch_attribute(self.target, "typedattr", 123, type_validation=False)
```

Comment:
I think we should also add a positive test case "It patches & works with valid type".